enhancements/oc/mirroring-release-signatures.md

```diff
@@ -0,0 +1,256 @@
+Success
+Update image:  openshift/release:4.3.0
+
+To upload local images to a registry, run:
+
+    oc image mirror --from-dir=mirror file://openshift/release:4.3.0* REGISTRY/REPOSITORY
+```
+
+creates an output filesystem like:
+
+```console
+$ tree mirror
+mirror
+└── v2
+    └── openshift
+        └── release
+            ├── blobs
+            │   ├── sha256:005337dc1f6870934cf5efaf443786149e6a71041a91566ce1e16e78bce511a9
+            ...
+            │   └── sha256:ffc132fefac522ec38b1b22ecd67a5173565efb3e8929e47ef7e0b4ee7920adf
+            └── manifests
+                ├── 4.3.0 -> sha256:3a516480dfd68e0f87f702b4d7bdd6f6a0acfdac5cd2e9767b838ceede34d70d
+                ...
+                ├── 4.3.0-thanos -> sha256:42ab3f59d3e769e88af89c30304b9cc77a76b66960747e2e1a6f8db097420858
+                ├── sha256:00ec58112784d340179e045bc70399ac1bd509ae43c4411e934ced23d016b2a1
+                ...
+                └── sha256:ffa87c992c5a2c51290e6d67cb609f29893dd147c31cff639ff6785ae7a1cfe2
+
+5 directories, 514 files
+```
+
+This filesystem can then be applied to the local registry with the given `oc image mirror` command.
+
+With this enhancement, the output format would gain an additional directory `config` under the base directory, a sibling to the current `v2`, containing configuration manifest files that can be applied to the local cluster.
+For this enhancement, the `config` directory will only create a single file, whose filename will be `signature-${ALGORITHM}-${ENCODED_PREFIX}.yaml`, including the algorithm and first sixteen characters of the encoded portion of the release image [digest][].
+The manifest file can be applied to the local cluster with:
+
+```console
+$ oc apply -f mirror/config/signature-sha256-81154f5c03294534.yaml
+```
+
+or similar (with the appropriate algorithm instead of `sha256` and encoded prefix instead of `81154f5c03294534`).
+
+### Pushing the release image signature ConfigMap manifest to disk while pushing containers to the mirror registry
+
+Users mirroring a release before creating a cluster may want to [push both container images and the release image signature ConfigMap manifest to disk](#pushing-to-disk).
+But they might also want to push container images directly to a mirror repository (as in [the *applying directly to the target cluster* case](#applying-directly-to-the-target-cluster)) while still pushing the release image signature ConfigMap to disk (because there is not yet a cluster into which that manifest can be pushed).
+While you could address this use-case with [the *pushing to disk* flow](#pushing-to-disk) followed by an immediate `oc image mirror ...`, it is more convenient to have `oc adm release mirror ...` push to the registry directly.
+Because setting `--to-dir` adjusts the image source, this enhancement extends the command with a new `--release-image-signature-to-dir` that can be used to set a configuration manifest output directory without setting a container image output directory.
```

Comment:
Ref prior comment https://github.com/openshift/enhancements/pull/188#discussion_r393735820
"...consistent with our naming patterns (--to-dir, --to-release)", so to adhere make this `--to-release-image-signature-dir`

Reply:
My [pushback against naming the directory][1] (and instead focusing on what we are pushing into the directory) was not convincing?  Expanding just in case, this option is about putting release image signatures in an unclassified directory, while `--to-release-image-signature-dir` sounds more like putting unspecified things in a directory that is specific to release image signatures.

[1]: https://github.com/openshift/enhancements/pull/188#discussion_r393812980

Reply:
I don't feel strongly about it or disagree with your pushback but rather am highlighting Clayton's comment to be sure you dodn;t overlook in this re-roll. Again, I just want to reach agreement so I can push my implementation :)